src/Mcamara/LaravelLocalization/LaravelLocalization.php

```diff
@@ -168,7 +169,10 @@ public function setLocale($locale = null)
         }
 
         $this->app->setLocale($this->currentLocale);
-
+        
+        // setLocale to use Carbon source locales. Enables diffForHumans() localized
+        Carbon::setLocale(config('app.locale'));
```

Comment:
It is module, and module shouldn't change something in other module cos it is not transperent, it is additional dependency etc. 

This module set application locale, but locales in other places shoud be set with this application locale, not directly from here. In this case it will be abstracted enough to keep flexibility.